benchmarks/clhqueuebench.c

```diff
@@ -40,15 +42,18 @@ inline static void enqueue(PoolStruct *pool_node, Object arg, int pid) {
 
 inline static Object dequeue(int pid) {
     Object result;
+    ListNode *n = NULL;
 
     CLHLock(lhead, pid);
     if (Head->next == null)
-        result = -1;
+        return -1;
     else {
-        result = Head->next->value;
```

Comment:
was that a bug?

Reply:
Yes it was a bug. 
I think its time to introduce a smoke test for the 1) stacks, 2) queues, and 3) Fetch&Multiply objects.